enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,149 @@
+
+Once OLM has generic support for Prometheus Operators, OLM should offer a path to automatically integrate Red Hat operators with the OpenShift Monitoring Prometheus Operator. There are a few requirements that must be met for the OpenShift Monitoring Prometheus Operator to scrape an application's metrics.
+
+##### Namespace Requirements
+The application must be deployed in a namespace that:
+
+- Is prefixed with `openshift-`
+- Is labeled with `openshift.io/cluster-monitoring=true`
+
+##### ServiceMonitor Requirements
+
+A ServiceMonitor object must be created in the namespace mentioned above and point to the application's metrics endpoint.
+
+>Note: This requirement is already fulfilled with generics support for the Prometheus Operator.
+
+##### RBAC Requirements
+
+A role/rolebinding must be created that provides the Prometheus Operator ServiceAccount with the appropriate RBAC privileges to discover the `ServiceMonitors` and `PrometheusRules` objects in the newly created namespace.
+
+#### Fulfilling Namespace Requirements
+
+As OLM gains support for `Namespace` objects, The `ClusterServiceVersion` will be updated to include an optional `NamespaceRequirements` object field. The `NamespaceRequirements` will look like the following:
+
+```golang
+type NamespaceRequirements struct {
+  //+optional
+  // name represents where the operator must be deployed.
+  name string
+
+  //+optional
+  // labels are labels that must be present on the given namespace.
+  labels []string
+}
+```
+
+#### Fulfilling Requirement 3: Updating the Prometheus Operator ServiceAccount
+
+In order for the Prometheus Operator to recognize take advantage of this new feature, its ServiceAccount must have the proper RBAC permissions to view events on the `ServiceMonitors` and `PrometheusRules` objects in the operator's namespace. OLM can simply update the default Prometheus instance with the correct RBAC privileges when the `openshift.io/cluster-monitoring=true` label is present on the namespace.
+
+#### Risks and Mitigations
+
+It is possible that the Prometheus team will not allow OLM managed operators to use the default prometheus operator to record metrics. If they are concerned about OLM managed operators overloading the prometheus instance, one solution that addresses this concern includes creating a new operator that is responsible for adding the `openshift.io/cluster-monitoring=true` label to namespaces and for generating the Telemter RBAC. This operator would:
+
+- Be deployed in the `openshift-monitoring` namespace
+- Have a list of approved OLM managed operators that may report metrics to the OpenShift Monitoring Prometheus Operator
+- Watch for events on namespaces
+- If an event occurs on a namespace that starts with `openshift-` and the namespace only includes valid operators, add the `openshift.io/cluster-monitoring=true` annotation and create RBAC that allows the OpenShift Monitoring Prometheus Operator service account to watch for events in the namespace
```

Comment:
Just to underline the above statement. With user-workload-monitoring no special labels have to be set on namespaces and no special RBAC permissions have to be granted to the "user-workload-monitoring prometheus operator".